core/shared/src/main/scala/zio/ZQueue.scala

```diff
@@ -186,7 +186,7 @@ sealed abstract class ZQueue[-RA, -RB, +EA, +EB, -A, +B] extends Serializable {
       def awaitShutdown: UIO[Unit] = self.awaitShutdown *> that.awaitShutdown
       def size: UIO[Int]           = self.size.zipWithPar(that.size)(math.max)
       def shutdown: UIO[Unit]      = self.shutdown.zipWithPar(that.shutdown)((_, _) => ())
-      def isShutdown: UIO[Boolean] = self.isShutdown
+      def isShutdown: UIO[Boolean] = self.isShutdown.zipWithPar(that.isShutdown)(_ && _)
```

Comment:
Actually should this be `&&` or `||`? You're going to get interrupted if you try to pull from a composite queue if either queue has been shutdown, so in that sense `||` seems like the wrong semantics. But on the other hand shutdown of the composite queue hasn't been completed until both queues have been shut down.